src/raftstore/store/store.rs

```diff
@@ -794,26 +794,44 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         let ids: Vec<u64> = self.pending_raft_groups.drain().collect();
         let pending_count = ids.len();
 
+        let mut ready_results: Vec<(u64, ReadyResult)> = Vec::with_capacity(ids.len());
         for region_id in ids {
-            let mut ready_result = None;
             if let Some(peer) = self.region_peers.get_mut(&region_id) {
-                match peer.handle_raft_ready(&self.trans, &mut self.raft_metrics) {
+                match peer.handle_raft_ready_append(&self.trans, &mut self.raft_metrics) {
                     Err(e) => {
                         // TODO: should we panic or shutdown the store?
-                        error!("{} handle raft ready err: {:?}", peer.tag, e);
+                        error!("{} handle raft ready append err: {:?}", peer.tag, e);
```

Comment:
I don't think it's safe to return here. If region A is pending for calling `on_ready_result`, and region B returns early, it may make everything go wrong.